src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/HistogramAdvice.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Contains settings used to advise metrics consumers how to construct storage for <see cref="Histogram{T}"/> instruments.
+    /// </summary>
+    /// <typeparam name="T">Histogram value type.</typeparam>
+    public sealed class HistogramAdvice<T> where T : struct
+    {
+        /// <summary>
+        /// Constructs a new instance of <see cref="HistogramAdvice{T}"/>.
+        /// </summary>
+        /// <param name="explicitBucketBoundaries">
+        /// <para>Explicit bucket boundaries advised to be used with the histogram.</para>
+        /// <para>Notes:
+        /// <list type="bullet">
+        /// <item>Bucket boundaries MUST be specified in ascending order and CANNOT contain duplicate values.</item>
+        /// <item>An empty set of bucket boundaries hints that the histogram should NOT contain buckets and should only track count and sum values.</item>
+        /// </list>
+        /// </para>
+        /// </param>
+        public HistogramAdvice(IEnumerable<T> explicitBucketBoundaries)
+        {
+            if (explicitBucketBoundaries is null)
+            {
+                throw new ArgumentNullException(nameof(explicitBucketBoundaries));
+            }
+
+            List<T> explicitBucketBoundariesCopy = [.. explicitBucketBoundaries];
```

Comment:
>List<T> explicitBucketBoundariesCopy = [.. explicitBucketBoundaries]; [](http://example.com/codeflow?start=12&length=69)

List<T> explicitBucketBoundariesCopy = new List<T>(explicitBucketBoundaries);

Reply:
Would it make sense to do it like:

`IReadOnlyList<T> explicitBucketBoundariesCopy =  new ReadOnlyCollection<T>(new List<T>(explicitBucketBoundaries));`

To ensure no-one can ever mutate it after creation?